src/sentry/integrations/slack/message_builder/notifications/base.py

```diff
@@ -22,15 +24,64 @@ def __init__(
         self.context = context
         self.recipient = recipient
 
-    def build(self) -> SlackBody:
+    def build(self) -> SlackAttachment | SlackBlock:
         callback_id_raw = self.notification.get_callback_data()
-        return self._build(
-            title=self.notification.build_attachment_title(self.recipient),
-            title_link=self.notification.get_title_link(self.recipient, ExternalProviders.SLACK),
-            text=self.notification.get_message_description(self.recipient, ExternalProviders.SLACK),
-            footer=self.notification.build_notification_footer(
-                self.recipient, ExternalProviders.SLACK
-            ),
-            actions=self.notification.get_message_actions(self.recipient, ExternalProviders.SLACK),
-            callback_id=json.dumps(callback_id_raw) if callback_id_raw else None,
+        title = self.notification.build_attachment_title(self.recipient)
+        title_link = self.notification.get_title_link(self.recipient, ExternalProviders.SLACK)
+        text = self.notification.get_message_description(self.recipient, ExternalProviders.SLACK)
+        footer = self.notification.build_notification_footer(
+            self.recipient, ExternalProviders.SLACK
+        )
+        actions = self.notification.get_message_actions(self.recipient, ExternalProviders.SLACK)
+        callback_id = json.dumps(callback_id_raw) if callback_id_raw else None
+        if not features.has("organizations:slack-block-kit", self.notification.organization):
+            return self._build(
+                title=title,
+                title_link=title_link,
+                text=text,
+                footer=footer,
+                actions=actions,
+                callback_id=callback_id,
+            )
+
+        first_block_text = ""
+        if title_link:
+            if title:
+                first_block_text += f"<{title_link}|*{escape_slack_text(title)}*>  \n"
+            else:
+                first_block_text += f"<{title_link}|*{escape_slack_text(title_link)}*>  \n"
+        elif title:  # ie. "ZeroDivisionError",
+            first_block_text += f"*{escape_slack_text(title)}*  \n"
+
+        if text:  # ie. "division by zero", comments
+            first_block_text += text
+
+        blocks = []
+        if first_block_text:
+            blocks.append(self.get_markdown_block(text=first_block_text))
+        if footer:
+            blocks.append(self.get_context_block(text=footer))
+
+        actions_block = []
+        for action in actions:
+            if (
+                action.label
+                in (
+                    "Archive",
+                    "Ignore",
+                    "Mark as Ongoing",
+                    "Stop Ignoring",
+                    "Turn on personal notifications",
+                )
+                or action.url
+            ):
+                actions_block.append(self.get_button_action(action))
+            elif action.name == "assign":
```

Comment:
all of these except "Turn on personal notifications" and `action.url` are not currently used in this pr, however, the pr that will follow up (which resolves #61460) will use these (i originally had implemented both in one pr; they have since been split but i kept this part)